### ISSUE TITLE
Colors from tokens in Token Studio and in Local Variable don’t match for the same token

### ISSUE THREAD

**Issue by mackoj** [tokens-studio/figma-plugin#2657]
**Issue Description**

In the `Local Variable` section of `Token Studio`, certain **colors** do not match those in the `Token Studio` itself.

**Steps to Reproduce**

1. Open "Token Studio".
2. Load your tokens.
3. Navigate to "Styles & Variables".
4. Create variables if none exist.
5. Access "Local variables".

When using multiple sets and themes, the resolver (the process that looks up token values) behaves differently in Local variables compared to Token Studio. Specifically, the colors displayed are inconsistent with those in Token Studio, resulting in incorrect variable assignments.

**Expected Behavior**

I expect the colors displayed in "Local variables" to be consistent with those in "Token Studio".

**Screenshots or Screencasts**

The color in `vp.web.component.button.color.primary.regular.container.default` in the theme `legacy-light-web` and `legacy-dark-web` is not valid in "Local variables" but it correct in "Token Studio".

The color in `vp.web.component.button.color.primary.inverse.content` in the theme `legacy-light-web` and `legacy-dark-web` is not valid in "Local variables" but it correct in "Token Studio".

The other colors in this excerpt seems to be valid.

https://github.com/tokens-studio/figma-plugin/assets/661647/e72f6512-33e4-4af5-b930-fb26f858e274

**Assessment**

It appears that this issue occurs when multiple sets (e.g., web, apple, android) are used. The resolver fails to correctly resolve colors based on the theme, sometimes using colors from the first theme only.

This functionality works correctly in Token Studio, indicating that the problem likely lies in how colors are resolved specifically within "Local variables".

I hope this information helps in identifying and resolving the issue.

**Related Tickets and Pull Request**

Here are some related tickets that appear to be addressing issues similar to this one. However, since I am uncertain if the root cause is identical, I have created this new issue:
- [Issue #2651](https://github.com/tokens-studio/figma-plugin/issues/2651)
- [Issue #2477](https://github.com/tokens-studio/figma-plugin/issues/2477)

Additionally, there is a pull request that could potentially fix a related issue. Unfortunately, I am unsure how to test the code from this PR in my Figma environment. Moreover, I cannot guarantee that it will resolve my specific issue:

- [Pull Request #2568](https://github.com/tokens-studio/figma-plugin/pull/2568)

**Comment by mackoj** [comment]
Hi @six7, sorry to bother you. Do you think this is a real bug or a user issue?

**Comment by SamIam4Hyma** [comment]
@mackoj have you been able to replicate this issue using any of the new beta builds of the plugin we are testing? would be good to have this info (:

**Comment by mackoj** [comment]
Yes, @SamIam4Hyma, I did reproduce it on the first beta, as stated in the original message and didn't test since.

**Comment by six7** [comment]
@mackoj can you share your tokens with us? Or the file - it's a bit hard to see from the video what values you used, trying to find the root cause of this

**Comment by mackoj** [comment]
@six7 can I send the token to you in MP ?

**Comment by SamIam4Hyma** [comment]
closing this issue as its stale.
